regtests/client/python/cli/command/principals.py

```diff
@@ -93,5 +111,34 @@ def execute(self, api: PolarisDefaultApi) -> None:
                 properties=new_properties
             )
             api.update_principal(self.principal_name, request)
+        elif self.principals_subcommand == Subcommands.ACCESS:
+            principal = api.get_principal(self.principal_name).to_dict()['name']
+            principal_roles = self._get_principal_roles(api)
+
+            # Initialize the result structure
+            result = {
+                'principal': principal,
+                'principal_roles': []
+            }
+            
+            # Construct the result structure for each principal role
+            for principal_role in principal_roles:
+                role_data = {
+                    'name': principal_role,
+                    'catalog_roles': []
+                }
+                # For each catalog role, get associated privileges
+                for catalog in self._get_catalogs(api):
```

Comment:
I wonder how frequently duplicates will appear, and whether we could try to remove them